src/main/java/com/minecolonies/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -487,49 +480,61 @@ private AIState requestMaterials()
             job.complete();
         }
 
-        //todo as soon as material handling has been implemented this should be set to work!
         //We need to deal with materials
-        if (!Configurations.builderInfiniteResources)
+        if (!Configurations.builderInfiniteResources && !job.getWorkOrder().isRequested())
         {
-            //TODO thread this
             while (job.getStructure().findNextBlock())
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS135)